File: 02_sorting_and_searching/02_apartments.py
n, m, k = map(int, input().split())
desired_apt_size = list(map(int, input().split()))
apt_size = list(map(int, input().split()))

desired_apt_size.sort()
apt_size.sort()
result_count = 0

i = j = 0
while i < len(desired_apt_size) and j < len(apt_size):
    if desired_apt_size[i] - k > apt_size[j]:
        j += 1
    elif desired_apt_size[i] + k < apt_size[j]:
        i += 1
    else:
        i += 1
        j += 1
        result_count += 1

print(result_count)

File: 02_sorting_and_searching/02_apartments.cpp
#include <bits/stdc++.h>
using namespace std;

void solve() {
    int n, m, k; 
    cin >> n >> m >> k;
    vector<int> desired_apt_size(n);
    vector<int> apt_size(m);
    for(int i=0; i<n; i++) {
        cin >> desired_apt_size[i];
    }
    for(int i=0; i<m; i++) {
        cin >> apt_size[i];
    }
    sort(desired_apt_size.begin(), desired_apt_size.end());
    sort(apt_size.begin(), apt_size.end());
    int count = 0; 

    int i = 0, j = 0; 
    while(i < n && j < m) {
        if(apt_size[j] < desired_apt_size[i] - k) 
            j++; 
        else if(apt_size[j] > desired_apt_size[i] + k)
            i++; 
        else {
            count++; 
            i++; 
            j++; 
        }
    }
    cout << count << endl;
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    solve();
}
